packages/e2e-tests/src/selectors.ts

```diff
@@ -330,29 +345,105 @@ export class Channel {
     return this.driver.findElement(By.xpath('//ul[@id="messages-scroll"]'))
   }
 
-  async messagesGroup() {
-    const messagesList = await this.messagesList
-    return await messagesList.findElement(By.css('li'))
-  }
-
-  async messagesGroupContent() {
-    const messagesGroup = await this.messagesGroup()
-    return await messagesGroup.findElement(By.xpath('//p[@data-testid="/messagesGroupContent-/"]'))
+  async waitForUserMessageByText(username: string, messageContent: string) {
+    logger.info(`Waiting for user "${username}" message "${messageContent}"`)
+    return this.driver.wait(async () => {
+      const messages = await this.getUserMessages(username)
+      for (const element of messages) {
+        const text = await element.getText()
+        logger.info(`Potential message with text: ${text}`)
+        if (text.includes(messageContent)) {
+          logger.info(`Found message with matching text ${text}`)
+          return element
+        }
+      }
+      throw logAndReturnError(`No message found for user ${username} and message content ${messageContent}`)
+    })
   }
 
-  async waitForUserMessage(username: string, messageContent: string) {
-    logger.info(`Waiting for user "${username}" message "${messageContent}"`)
+  async waitForUserMessageByFilename(
+    username: string,
+    filename: string,
+    fileType: UploadedFileType
```

Comment:
No timeout param? `this.driver.wait` allows for specifying a timeout parameter to fail if the element does not resolve in time. We should take advantage of that so that we don't end up in a perpetual wait instead of just properly failing the test.

Reply:
Blanket response: All of the finds/visibility checks (should) now have timeouts and in most cases they aren't using a parameter in the wrapper function.  We can make a ticket to update that but I didn't want to add even more time to this.